client/src/test/java/org/asynchttpclient/ClientStatsTest.java

```diff
@@ -0,0 +1,56 @@
+package org.asynchttpclient;
+
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.asynchttpclient.Dsl.asyncHttpClient;
+import static org.asynchttpclient.Dsl.config;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Created by grenville on 9/25/16.
+ */
+public class ClientStatsTest extends AbstractBasicTest {
+
+    @Test(groups = "standalone")
+    public void testClientStatus() throws Throwable {
+        try (final DefaultAsyncHttpClient client = (DefaultAsyncHttpClient) asyncHttpClient(config().setKeepAlive(true).setPooledConnectionIdleTimeout(1000))) {
+            final String url = getTargetUrl();
+
+            final ClientStats emptyStats = client.getClientStats();
+
+            assertEquals(emptyStats.getActiveConnectionCount(), 0);
+            assertEquals(emptyStats.getIdleConnectionCount(), 0);
+            assertEquals(emptyStats.getTotalConnectionCount(), 0);
+            assertEquals(emptyStats.toString(), "There are 0 total connections, 0 are active and 0 are idle.");
+
+            final List<ListenableFuture<Response>> futures = new ArrayList<>();
+            for (int i = 0; i < 5; i++) {
+                logger.info("{} requesting url [{}]...", i, url);
+                futures.add(client.prepareGet(url).execute());
+            }
+
+            for (ListenableFuture<Response> future : futures) {
+                future.get();
+            }
```

Comment:
Have a look at `EchoHandler`, which is the default handler used in the Jetty servers.
It can use a `LockThread` header to block the response on the server side. It current performs a hard coded 40s sleep, but it's actually no longer used. It could be used to pass the sleep value instead.